whd/dispatcher.py

```diff
@@ -213,6 +219,130 @@ def _set_pr_labels(self, pr_event, resources):
                     f" in '{repository_path}' found. Ignoring, since they are not an org member'."
                 )
 
+    def _resource_versions_and_jobs(self, concourse_api, matching_resources, pr_event):
+        resource = [r for r in matching_resources if r.type == "pull-request"]
+        if len(resource) != 1:
```

Comment:
firstly: please handle _both_ unexpected error cases. otherwise, this will make debugging unnecessarily hard!
secondly: why would it be an error or unexpected to find more than one resource?
thirdly: why would it be unexpected to find none?

ALSO: after reading through `dispatcher.py#_matching_resources()`, I found that:

1. resources are already filtered by type. so why do the filtering again? This is confusing, and unnecessary!
2. resources are already being filtered by repository. so again: why filter again. why validate again? are there reasons to believe that said method is erroneous? In that case, please fix that one, instead of (seemingly) working around a bug there (I would not be aware that there is a bug, tbh)